src/index.ts

```diff
@@ -33,10 +46,58 @@ async function serverChannel(channel: Channel, { projectToken }: { projectToken:
     });
   });
 
+  channel.on(
+    UPDATE_PROJECT,
+    async ({ projectId, projectToken: updatedProjectToken }: UpdateProjectPayload) => {
+      projectToken = updatedProjectToken;
+
+      const mainPath = await findConfig("main");
+      const MainConfig = await readConfig(mainPath);
+
+      const addonsConfig = MainConfig.getFieldValue(["addons"]);
+      const updatedAddonsConfig = addonsConfig.map(
+        (addonConfig: string | { name: string; options?: Record<string, string> }) => {
+          const fullConfig = typeof addonConfig === "string" ? { name: addonConfig } : addonConfig;
+          if (fullConfig.name === CHROMATIC_ADDON_NAME) {
+            return {
+              ...fullConfig,
+              options: { projectId, projectToken, ...fullConfig.options },
+            };
+          }
+          return addonConfig;
+        }
+      );
+
+      MainConfig.setFieldValue(["addons"], updatedAddonsConfig);
+      await writeConfig(MainConfig);
+    }
+  );
+
   return channel;
 }
```

Comment:
I'd expect a 
```
channel.on(
    REQUEST_PROJECT,
```
...here as well, that allows the front-end to request/receive the projectId

Reply:
Not currently implemented. We just read it on start up from env, and if/when the user picks a new project.

Reply:
With the code above, it can't be super difficult to implement reading the value, right?

Reply:
Why would we need to read the value? We already get it from the preset function.

Reply:
Because: the preset function will get a stale value. We don't re-read the files when calling the preset function.